R/init-package.R

```diff
@@ -2,37 +2,107 @@
 #'
 #' This will be called once when the package is loaded
 #'
+#'
 #' @import rSharp
 #' @keywords internal
 .initPackage <- function() {
+  # Get library directory path once
+  libDir <- system.file("lib", package = ospsuiteEnv$packageName)
+
+  # Helper function for file paths
   filePathFor <- function(name) {
-    system.file("lib", name, package = ospsuiteEnv$packageName)
+    file.path(libDir, name)
   }
 
-  # Make .dll binaries available on windows by extending PATH
+  # Setup platform-specific SQLite DLL
+  targetDll <- file.path(libDir, "System.Data.SQLite.dll")
+  sourceDll <- file.path(
+    libDir,
+    if (Sys.info()[["sysname"]] == "Darwin") {
+      "System.Data.SQLite.mac.dll"
+    } else {
+      "System.Data.SQLite.others.dll"
```

Comment:
interesting file name :)

Reply:
Naming is one of the hardest challenges in sw-dev 😆 
But maybe something like       "System.Data.SQLite.windows_linux.dll" ?